CONTRIBUTING.adoc

```diff
@@ -0,0 +1,184 @@
+The goal of adding new content is to get technically correct documentation into the repo before it is lost. Tech Writers may provide some quick guidance on getting documentation into the correct location, but won't be providing a detailed list of items to change.
+
+=== Refining and Reviewing Content
+
+Once the raw documentation has made it into the repo, tech writers and other communications experts will review and revise the documentation to make it easier to consume. This will be done as a second PR; it's often easier for the tech writers to clean up or rewrite a section of prose than to teach an engineer what to do, and most engineers would prefer to get back to working on code.
+
+When revising the content, the tech writer will create a new PR and send it to the original author to ensure that the content is technically correct; they may also ask a few clarifying questions, or add details such as diagrams or notes if needed. It's not necessarily expected that tech writers will actually execute the steps of a tutorial -- it's expected that the SME is responsible for a working tutorial or how-to.
+
+== Tech Writer Guide
+
+
+=== Type of documentation
+
+Keep in mind the audience (Developers or Administrators) and type of document when organizing and reviewing documentation. See https://documentation.divio.com/ for a more in-depth discussion of documentation types.
+
+=== Documentation Structure
+
+TODO: link to intended documentation layout. A general warning about https://en.wikipedia.org/wiki/Conway%27s_law[Conway's Law]: documents will naturally tend to be distributed by team that produced them. Try to fight this, and organize documents based on where the _reader_ will look for them. (i.e. all tutorials together, maybe with indications as to which components they use, rather than putting all serving tutorials in one place)
+
+In some cases, the right place for a document may not be on the docs website -- a blog post, documentation within a code repo, or a vendor site may be the right place. Be generous with offering to link to such locations; documents in the main documentation come with an ongoing cost of keeping up to date.
+
+=== Branches
+
+Knative attempts to https://github.com/knative/community/blob/master/mechanics/RELEASE-VERSIONING-PRINCIPLES.md[support the last 4 releases]. By default, new documentation should be written on the `master` branch and then cherry-picked to the release branches if needed. Note that the default view of https://knative.dev/ is of the most recent release branch, which means that changes to `master` don't show up unless explicitly cherrypicked. This also means that documentation changes for a release _should be made during the development cycle_, rather than at the last minute or after the release.
+
+The https://github.com/kubernetes/test-infra/tree/master/prow/external-plugins/cherrypicker[`/cherrypick` tool] can be use to automatically pull back changes from `master` to previous releases if necessary.
```

Comment:
This is useful information, I didn't know this and have been manually cherrypicking when I had to do it.

Reply:
Yeah, I tried to collect the process/tools items that seemed to have been missed in one place for reference. Documenting the documentation process. 😁